src/v2i-hub/PedestrianPlugin/src/PedestrianPlugin.cpp

```diff
@@ -132,16 +132,15 @@ int PedestrianPlugin::checkXML()
 			PLOG(logDEBUG) << "flir session not yet initialized: " << std::endl;
 		}
 		else
-		{
-			std::string currentXML = flirSession->getPSMXML();
-
-			if (currentXML.compare(lastGeneratedXML) != 0)
-			{				
-				PLOG(logINFO) << "CheckXML: Broadcasting new CP PSM!" << std::endl;
-
-				BroadcastPsm(const_cast<char*>(currentXML.c_str()));
-				lastGeneratedXML = currentXML;
-			}
+		{	
+			//retrieve the PSM queue and send each one to be broadcast, then pop		
+			std::queue<string> &currentPSMQueue = flirSession->getPSMQueue();
+
+			for (int i = 0; i < currentPSMQueue.size(); i++)
```

Comment:
You will need to lock the mutex here because the one in get method is out of scope and your lock would have been released.

Reply:
I am actually going to merge this now because on second thought I don't think you need may not need locks.  Lets test and see.